### ISSUE TITLE
Architecture diagram

### ISSUE THREAD

**Issue by Aderfish** [ProximaEPFL/proxima#56]

**Comment by gruvw** [comment]
I created, drew and latter converted to the Penpot (see "Architecture Diagram" page) the Architecture Diagram for the project.

I included everything that we coded already and the `comments` + `profile` parts that will probably be added during milestone 2 (see with PO).

![Image](https://github.com/ProximaEPFL/proxima/assets/63407038/ce7aefae-9354-4f6c-bced-2165c0190417)

I don't think it makes so much sense to go deeper into the application architecture yet, as there is a lot of things we still need to decide/settle with (especially challenges and groups).

See on MS1 deliverables:
> It is still early, so we don't expect you to deliver a finalized architecture diagram for M1. 

I think it is a good idea not to waste too much time on the architecture diagram of the future features before we give it a proper thought + ER diagrams as it might change quite a lot till then.
Anyway, this diagram still gives a proper quick reference understanding of the whole application architecture, which was the intended purpose of this deliverable 👍
